File: EmployeesServer/Employees.API/Models/EmployeePostModel.cs
using Employees.Core.DTOs;
using Employees.Core.Entities;

namespace Employees.API.Models
{
    public class EmployeePostModel
    {
        
        public string FirstName { get; set; }
        public string Surname { get; set; }
        public string IdentityNumber { get; set; }
        public string Gender { get; set; } //Male / Female
        public DateTime DateOfBirth { get; set; }
        public DateTime BeginningOfWork { get; set; }
       
      
        public List<EmployeePositionPostModel> Positions { get; set; }
    }
}

File: employee_client/src/app/models/employee-post.model.ts
import { EmployeePosition } from "./employee-position.model";

export class EmployeePost {
    firstName: string;
    surname: string;
    identityNumber: string;
    gender: string; // Assuming EGender is defined elsewhere (see note below)
    dateOfBirth: Date;
    beginningOfWork: Date;
    positions?: EmployeePosition[]; // Optional array of EmployeePosition objects
  } 
  // export enum EGender {
  //   Male,
  //   Female
  // }
